tracer/src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -26,16 +27,27 @@ public static class SpanExtensions
         /// <param name="samplingPriority">The new sampling priority for the trace.</param>
         public static void SetTraceSamplingPriority(this ISpan span, SamplingPriority samplingPriority)
         {
-            span.SetTraceSamplingPriority((int)samplingPriority);
+            if (span == null) { ThrowHelper.ThrowArgumentNullException(nameof(span)); }
+
+            if (span.Context is SpanContext { TraceContext: { } traceContext })
+            {
+                traceContext.SetSamplingPriority((int)samplingPriority, SamplingMechanism.Manual);
+            }
         }
 
-        internal static void SetTraceSamplingPriority(this ISpan span, int samplingPriority)
+        /// <summary>
+        /// Sets the sampling priority for the trace that contains the specified <see cref="ISpan"/>.
+        /// </summary>
+        /// <param name="span">A span that belongs to the trace.</param>
+        /// <param name="priority">The new sampling priority for the trace.</param>
+        /// <param name="mechanism">The new sampling mechanism for the trace.</param>
+        internal static void SetTraceSamplingDecision(this ISpan span, int priority, int? mechanism = null)
```

Comment:
nit: do we need the default `null` for mechanism here? Can we just make it required? (I much prefer having two overloads if the answer is no, but I think we disagree on that 😛)

Reply:
Remove, thanks!